packages/__tests__/2-runtime/au-slot.spec.ts

```diff
@@ -0,0 +1,1015 @@
+    );
+
+    yield new TestData(
+      'supports accessing inner scope with $host',
+      `<my-element> <div au-slot="s2">\${message}</div> <div au-slot="s1">\${$host.message}</div> </my-element>`,
+      [
+        CustomElement.define(
+          { name: 'my-element', isStrictBinding: true, template: `<au-slot name="s1">s1</au-slot> <au-slot name="s2">s2</au-slot>` },
+          class MyElement {
+            public readonly message: string = 'inner';
+          }
+        ),
+      ],
+      { 'my-element': `<div>inner</div> <div>root</div>` },
+    );
+    // #endregion
+
+    // #region template controllers
+    {
+      @customElement({ name: 'my-element', isStrictBinding: true, template: `static <au-slot>default</au-slot> <au-slot name="s1" if.bind="showS1">s1</au-slot> <au-slot name="s2">s2</au-slot>` })
+      class MyElement {
+        @bindable public showS1: boolean = true;
+      }
+      yield new TestData(
+        'works with template controller - if',
+        `<my-element show-s1.bind="false"> <div au-slot="s2">p20</div> <div au-slot="s1">p11</div> <div au-slot="s2">p21</div> <div au-slot="s1">p12</div> </my-element>`,
+        [
+          MyElement,
+        ],
+        { 'my-element': `static default <div>p20</div><div>p21</div>` },
+        async function ({ host, scheduler }) {
+          const el = host.querySelector('my-element');
+          const vm: MyElement = CustomElement.for<Element, MyElement>(el).viewModel;
+
+          vm.showS1 = true;
+          await scheduler.yieldAll(10);
+
+          assert.html.innerEqual(el, `static default <div>p11</div><div>p12</div> <div>p20</div><div>p21</div>`, 'my-element.innerHTML');
+        },
+      );
+    }
+
+    {
+      @customElement({ name: 'my-element', isStrictBinding: true, template: `static <au-slot>default</au-slot> <au-slot name="s1" if.bind="showS1">s1</au-slot> <au-slot else name="s2">s2</au-slot>` })
+      class MyElement {
+        @bindable public showS1: boolean = true;
+      }
+      yield new TestData(
+        'works with template controller - if-else',
+        `
+        <my-element show-s1.bind="false"> <div au-slot="s2">p21</div> <div au-slot="s1">p11</div> </my-element>
+        <my-element show-s1.bind="true" > <div au-slot="s2">p22</div> <div au-slot="s1">p12</div> </my-element>
+        `,
+        [
+          MyElement,
+        ],
+        { 'my-element': `static default <div>p21</div>`, 'my-element+my-element': `static default <div>p12</div>` },
+        async function ({ host, scheduler }) {
+          const el1 = host.querySelector('my-element');
+          const el2 = host.querySelector('my-element+my-element');
+          const vm1: MyElement = CustomElement.for<Element, MyElement>(el1).viewModel;
+          const vm2: MyElement = CustomElement.for<Element, MyElement>(el2).viewModel;
+
+          vm1.showS1 = !vm1.showS1;
+          vm2.showS1 = !vm2.showS1;
```

Comment:
I think there should be an assertion like this, but before the values get reverted as well.
```suggestion
          assert.html.innerEqual(el1, `static default <div>p21</div>`, 'my-element.innerHTML');
          assert.html.innerEqual(el2, `static default <div>p12</div>`, 'my-element+my-element.innerHTML');

          vm1.showS1 = !vm1.showS1;
          vm2.showS1 = !vm2.showS1;
```

Reply:
It's already there in the line before the `async function` (currently L228) :)